src/com/amazon/corretto/crypto/provider/EdGen.java

```diff
@@ -2,19 +2,31 @@
 // SPDX-License-Identifier: Apache-2.0
 package com.amazon.corretto.crypto.provider;
 
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
 import java.security.KeyPair;
 import java.security.KeyPairGeneratorSpi;
+import java.security.PrivateKey;
+import java.security.PublicKey;
 import java.security.SecureRandom;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
 
 class EdGen extends KeyPairGeneratorSpi {
   /** Generates a new Ed25519 key and returns a pointer to it. */
   private static native long generateEvpEdKey();
 
   private final AmazonCorrettoCryptoProvider provider_;
+  private final KeyFactory kf;
 
   EdGen(AmazonCorrettoCryptoProvider provider) {
     Loader.checkNativeLibraryAvailability();
     provider_ = provider;
+    try {
+      kf = KeyFactory.getInstance("EdDSA", "SunEC");
```

Comment:
I'm a bit concerned about the FIPS implications. It looks like you're trying to export the KeyPair using JDK based interfaces by tumbling our Key classes into the SunEC KeyFactory. This helps with the interop story, but like we covered earlier there's a whole lot more work to get full interop working and doing it piecemeal can be counterproductive. Since this is along the lines of ASN.1 encode/decode it's probably okay, but it does break down our "all crypto comes from AWS-LC" story. Would like to get some more consensus from the team on how we should handle this.

Reply:
* There are places that ACCP relies on other providers for ASN.1 encode/decode, like [GCM parameters](https://github.com/corretto/amazon-corretto-crypto-provider/blob/main/src/com/amazon/corretto/crypto/provider/AesGcmSpi.java#L281).
* Without this, one cannot use keys generated by ACCP to perform sign/verify with another provider.
* It is expected that the private (public) key generated by ACCP can be type cast to [EdECPrivateKey](https://docs.oracle.com/en/java/javase/17/docs//api/java.base/java/security/interfaces/EdECPrivateKey.html) ([EdECPublicKey](https://docs.oracle.com/en/java/javase/17/docs//api/java.base/java/security/interfaces/EdECPublicKey.html)).

Reply:
I'm in favor of keeping this change to increase compatibility. This issue was accidentally caught by one of our own  basic cross-compatibility unit tests between SunEC and ACCP. Any existing Ed25519 users coming from SunJCE who are thinking about migrating to ACCP will likely add their own similar cross-compatibility unit test, and could easily run into this exact same issue themselves. It's better for ACCP to solve it once, than have many customers re-discover and re-solve this issue themselves.

Reply:
Given likely use-cases for Ed25519 signature key generation, I agree that compatibility/user experience outweigh performance here. We can optimize the latter in the future dependent on customer need.